pkg/ipamd/rpc_handler.go

```diff
@@ -120,6 +120,10 @@ func (s *server) AddNetwork(ctx context.Context, in *rpc.AddNetworkRequest) (*rp
 		}
 	}
 	if addr == "" {
+		if in.ContainerID == "" {
+			log.Errorf("ContainerID is empty")
+			return &failureResponse, nil
+		}
```

Comment:
With the code changes in  scripts/entrypoint.sh, i guess we wouldn't need this. If you still want to keep this sanity check , should we also account for validity of other fields that form the key ?

Reply:
Handled the checks for all fields in the IPAMKey